src/core/data/allocator.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "core/data/buffer.h"
+
+#include <unordered_map>
+
+namespace legate {
+
+class IntraTaskAllocator {
+ public:
+  using ByteBuffer = Buffer<int8_t>;
+
+ public:
+  IntraTaskAllocator() = default;
+  IntraTaskAllocator(Legion::Memory::Kind kind);
+  virtual ~IntraTaskAllocator();
+
+ public:
+  typedef char value_type;
+  char* allocate(size_t bytes);
+  void deallocate(char* ptr, size_t n);
```

Comment:
Is this the signature that Thrust expects? Otherwise, I think we should change it to `void (void*, size_t)`

Reply:
Yes, this is what thrust expects as far as i understand. I did not get it to work with void

Reply:
Ok. That still isn't a good excuse to use `char*`, as this isn't just meant for Thrust and using `char*` will cause trouble in other contexts. I think you should change it to `void*` and make an adaptor/wrapper to appease Thrust.

Reply:
I removed all type specifications from the generic allocator. The consumer now has to specify a small wrapper class that derives from it and adds the types.